js/app.js

```diff
@@ -1,23 +1,75 @@
-const btnInvestVariables = document.getElementById("btnInvestVariables");
-let investTime, propertySize, pricePerM, priceProperty, taxPCC, sourcingCost;
-
-const COURT_COST = 779;
-const NOTARY_COST = 1200;
-
-function setInvestmentVariables() {
-  investTime = parseInt(document.getElementById("investTime").value);
-  propertySize = parseInt(document.getElementById("propertySize").value);
-  priceProperty = parseInt(document.getElementById("priceProperty").value);
-  setVariableRelatedValues();
-}
-
-function setVariableRelatedValues() {
-  pricePerM = priceProperty / propertySize;
-  document.querySelector("#pricePerM > span").innerHTML = pricePerM.toFixed(1);
-  document.querySelector("#courtCosts > span").innerHTML = COURT_COST;
-  taxPCC = priceProperty * 0.02;
-  document.querySelector("#taxPCC > span").innerHTML = taxPCC;
-  document.querySelector("#notaryCost > span").innerHTML = NOTARY_COST;
-}
-
-btnInvestVariables.addEventListener("click", setInvestmentVariables);
+const investForm = document.querySelector("#investForm");
+
+const mapFormValues = (allFields, field) => {
+  const [name, value] = field;
+
+  return { ...allFields, [name]: +value };
+};
+
+const readForm = form => [...new FormData(form)].reduce(mapFormValues, {});
+
+const calculateCosts = userData => {
+  const calculatedCosts = {};
+  const pccTaxValueInPolen = 0.02;
+  const maklerProvisionPercentage = userData.maklerProvision / 100;
+  calculatedCosts.maklerProvisionCosts = maklerProvisionPercentage * userData.propertyPrice;
+  calculatedCosts.taxPCCCosts = userData.propertyPrice * pccTaxValueInPolen;
+
+  calculatedCosts.pricePerMeter = parseInt(userData.propertyPrice / userData.propertySize);
+  calculatedCosts.renovationCosts = userData.renovationCostsPerMonth * userData.propertySize;
+
+  calculatedCosts.rentCosts = userData.rentCostsPerMonth * userData.investTime;
+  calculatedCosts.electricityCosts = userData.electricityCostsPerMonth * userData.investTime;
+  calculatedCosts.waterCosts = userData.waterCostsPerMonth * userData.investTime;
+  calculatedCosts.heatingCosts = userData.heatingCostsPerMonth * userData.investTime;
+  calculatedCosts.insuranceCosts = userData.insuranceCostsPerMonth * userData.investTime;
+  return { ...userData, ...calculatedCosts };
```

Comment:
![picard-facepalm](https://user-images.githubusercontent.com/39776/71064271-fcd29600-216e-11ea-83f2-f35624a84969.jpg)

```suggestion
  return calculatedCosts;
```

Reply:
I wanna connect them here why I cant? :)

Reply:
We talked about that a lot already. 
The method is supposed to return `calculatedCosts` not combined entry values and `calculatedCosts`